File: client-js/evomaster-client-js/src/instrumentation/shared/TaintInputName.ts
export class TaintInputName {

    private static readonly PREFIX = "evomaster_";

    private static readonly POSTFIX = "_input";

    //Pattern.compile("\\Q"+PREFIX+"\\E\\d+\\Q"+POSTFIX+"\\E");
    private static regex = TaintInputName.PREFIX + "\\d+" + TaintInputName.POSTFIX

    private static partialMatch = new RegExp(TaintInputName.regex);

    private static fullMatch = new RegExp("^" + TaintInputName.regex + "$");

    /**
     * Check if a given string value is a tainted value
     */
    static isTaintInput(value: string): boolean {
        if (!value) {
            return false;
        }

        return TaintInputName.fullMatch.test(value);
    }


    static includesTaintInput(value: string): boolean {
        if (!value) {
            return false;
        }

        return TaintInputName.partialMatch.test(value);
    }


    /**
     * Create a tainted value, with the input id being part of it
     */
    static getTaintName(id: number): string {
        if (id < 0) {
            throw Error("Negative id");
        }
        /*
            Note: this is quite simple, we simply add a unique prefix
            and postfix, in lowercase.
            But we would not be able to check if the part of the id was
            modified.
         */
        return TaintInputName.PREFIX + id + TaintInputName.POSTFIX;
    }
}

File: client-dotnet/src/EvoMaster.Instrumentation_Shared/TaintInputName.cs
using System;
using System.Text.RegularExpressions;
using EvoMaster.Client.Util.Extensions;

namespace EvoMaster.Instrumentation_Shared {
    public class TaintInputName {
        /*
            WARNING:
            the naming here has to be kept in sync in ALL implementations of this class,
            including Java, JS and C#
         */

        private const string Prefix = "_EM_";

        private const string Postfix = "_XYZ_";

        private static readonly Regex Pattern = new Regex($"{Regex.Escape(Prefix)}\\d+{Regex.Escape(Postfix)}", RegexOptions.IgnoreCase);

        static TaintInputName(){
            
        }


        /// <summary>
        /// Check if a given string value is a tainted value
        /// </summary>
        /// <param name="value"></param>
        /// <returns></returns>
        public static bool IsTaintInput(string value) => value != null && Pattern.IsEntirelyMatch(value);


        public static bool IncludesTaintInput(string value) => value != null && Pattern.IsMatch(value);


        /**
     * Create a tainted value, with the input id being part of it
     */
        public static string GetTaintName(int id) {
            if (id < 0) {
                throw new ArgumentException("Negative id");
            }

            /*
                Note: this is quite simple, we simply add a unique prefix
                and postfix, in lowercase.
                But we would not be able to check if the part of the id was
                modified.
             */
            return Prefix + id + Postfix;
        }

        /**
     * One problem when using this type of taint, is that there can be constraints on the length
     * of the strings... and the taint value might end up being longer than it :-(
     * Not sure if there is really any simple workaround... but hopefully should be
     * so rare that we can live with it
     */
        public static int GetTaintNameMaxLength() {
            return Prefix.Length + Postfix.Length + 6;
        }
    }
}